app/models/case.rb

```diff
@@ -16,8 +16,43 @@ def complete
               mapping: %i[hearing_court_name name],
               constructor: :find_by_name, allow_nil: true
 
+  def complete?
+    case_hearing_attributes_present? &&
+      case_concluded_attributes_present? &&
+      client_remanded_attributes_present? &&
+      preorder_work_attributes_present?
+  end
+
   private
 
+  def case_hearing_attributes_present?
+    values_at(:hearing_court_name, :hearing_date).all?(&:present?)
+  end
+
+  def case_concluded_attributes_present?
+    if has_case_concluded.nil? || has_case_concluded == YesNoAnswer::NO.to_s
```

Comment:
Fine for now, but I think it would be great to add our attribute types to these active record objects. I'll explore that in a separate PR.